File: app/src/main/java/com/ladsoft/bakingapp/mvp/RecipesMvp.java
package com.ladsoft.bakingapp.mvp;

import com.ladsoft.bakingapp.entity.Recipe;

import java.util.List;

public interface RecipesMvp {
    interface View {
        void showRefresh(boolean show);
        void onRecipesLoaded(List<Recipe> recipes);
        void onRecipeLoadError();
    }

    interface Model {
        void setListener(Callback listener);
        void detachRecipesModelListener();
        void loadRecipes();

        interface Callback {
            void onDataLoaded(List<Recipe> recipes);
            void onDataLoadError(List<Recipe> recipes);
            void onDataLoadError();
        }
    }
}

File: app/src/main/java/com/ladsoft/bakingapp/mvp/RecipesMvp.kt
package com.ladsoft.bakingapp.mvp

import com.ladsoft.bakingapp.entity.Recipe


interface RecipesMvp {
    interface View {
        fun showRefresh(show : Boolean)
        fun onRecipesLoaded(recipes: MutableList<Recipe>?)
        fun onRecipeLoadError()
    }

    interface Model {
        fun setListener(listener: Callback)
        fun detachRecipesModelListener()
        fun loadRecipes()

        interface Callback {
            fun onDataLoaded(recipes: MutableList<Recipe>)
            fun onDataLoadError(recipes: MutableList<Recipe>)
            fun onDataLoadError()
        }
    }
}